### DISCUSSION TITLE
Android Appears to Be Killing App That Doesn't have High Memory Usage

### DISCUSSION THREAD

**Discussion by zburlott4** [airsdk/Adobe-Runtime-Support#3113] [General]
Adbt logcat is showing that the lmkd (low memory killer daemon) of android is selecting our app as the one one to terminate.   https://source.android.com/docs/core/perf/lmkd

Is it unusual for lmkd to be killing an app using 400mb of memory? 

The following line indicates android is ready to reclaim memory and its selecting our app is the one to be killed. **oom_adj 0** indicates that the app is running in the foreground as per android specifications.

`03-11 11:22:46.168   521   521 I lmkd    : Reclaim 'air.XXXXXXX' (23313), uid 10292, oom_adj 0, state 2 to free 419088kB`

Memory usage is only around 400 mb on an 4 Gg ram tablet. 

`03-11 11:22:46.163   521   521 I lmkd    : cached 0, sandbox(not0) 0
03-11 11:22:46.163   521   521 I lmkd    : skip proc : 7800 in heaviest 
03-11 11:22:46.163   521   521 I lmkd    : skip proc : 7815 in heaviest 
03-11 11:22:46.163   521   521 I lmkd    : skip proc : 7917 in heaviest 
03-11 11:22:46.163   521   521 I lmkd    : skip proc : 7953 in heaviest 
03-11 11:22:46.164   521   521 I lmkd    : skip proc : 7992 in heaviest 
03-11 11:22:46.165  2398  5028 D LeAppInfo: removeLeacReportedServerApp, appName: android.uid.system
03-11 11:22:46.166  2398  2398 I [MCFSDK_bluetooth_1.0.2109]_McfContext: onServiceDisconnected - -
03-11 11:22:46.167  2398  2398 D GuestManager: setMcfAdapter false
03-11 11:22:46.167  2398  2398 D HostManager: setMcfAdapter false
03-11 11:22:46.167  1136  1364 I ActivityManager: Changes in 10137 5 to 2, 0 to 7 Caller=com.android.server.am.OomAdjuster.updateOomAdjLocked:402 com.android.server.am.OomAdjuster.updateOomAdjLocked:449 com.android.server.am.ActivityManagerService.updateOomAdjLocked:21664 com.android.server.am.ProcessRecord.updateProcessInfo:1636 com.android.server.am.ProcessRecord.onStartActivity:1668 com.android.server.wm.-$$Lambda$VY87MmFWaCLMkNa2qHGaPrThyrI.accept:27 com.android.internal.util.function.pooled.PooledLambdaImpl.doInvoke:324 
03-11 11:22:46.167  1136  1364 I AppOps  : updateUidProcState uid: 10137 procState: 2 capability:7
03-11 11:22:46.168   521   521 I lmkd    : Reclaim 'air.XXXXXXX' (23313), uid 10292, oom_adj 0, state 2 to free 419088kB
03-11 11:22:46.169   521   521 I lmkd    : Reclaimed 419088kB at oom_adj 0`

**Comment by ajwfrost** [comment]
I'm curious to know whether lmkd had previously killed off other low-priority processes/apps?

I suspect it's targeting the AIR app not because of the the total memory usage there, but maybe there's something else that was triggering it. E.g. a single significant memory request, or some other situation where the app is hammering out memory requests for some reason...

What device / OS version was this running on? and how reproducible is it? Could you try reproducing it with an app that just allocates a 400MB byte array and then sits there doing nothing much else?
